packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.java

```diff
@@ -150,7 +150,7 @@ public void onHostDestroy() {
   private Uri contentUriFromFile(File file) {
     try {
       Application application = mModuleRegistry.getModule(ActivityProvider.class).getCurrentActivity().getApplication();
-      return FileProvider.getUriForFile(application, application.getPackageName() + ".provider", file);
+      return FileProvider.getUriForFile(application, application.getPackageName() + ".MailComposerFileProvider", file);
```

Comment:
In general, authorities are named in package-like style. So I would go with `.mailcomposer.provider` both here and in `AndroidManifest.xml`

But of course, that should fix the issue, so feel free to merge as it is.

Reply:
ah okay, if that's the precedent then I agree we should do that. Do you think the same changes should be made to other packages using authorities the same way? (e.g. [expo-sharing](https://github.com/expo/expo/tree/master/packages/expo-sharing))

Reply:
Thanks a lot for that!

I am a little worried about backward compatibility. I am almost (!) sure it won't break it, but would rather test i before. So maybe it is better to leave it as it is, than changed without testing.